cc/algorithms/partition-selection-strategies_test.cc

```diff
@@ -0,0 +1,373 @@
+	PreaggPartitionSelection::Builder test_builder;
+  	auto failed_build = test_builder.SetEpsilon(8.0)
+  									.SetMaxPartitionsContributed(9).Build();
+  	EXPECT_THAT(failed_build.status().code(),
+                Eq(base::StatusCode::kInvalidArgument));
+   	std::string message(std::string(failed_build.status().message()));
+  	EXPECT_THAT(message, MatchesRegex("^Delta has to be set.*"));
+}
+
+TEST(PartitionSelectionStrategiesTest, PreaggPartitionSelectionNotFiniteDelta) {
+	PreaggPartitionSelection::Builder test_builder;
+  	auto failed_build = test_builder.SetEpsilon(1.2).SetDelta(NAN)
+  									.SetMaxPartitionsContributed(3).Build();
+  	EXPECT_THAT(failed_build.status().code(),
+                Eq(base::StatusCode::kInvalidArgument));
+   	std::string message(std::string(failed_build.status().message()));
+  	EXPECT_THAT(message, MatchesRegex("^Delta has to be finite.*"));
+}
+
+TEST(PartitionSelectionStrategiesTest, PreaggPartitionSelectionInvalidDelta) {
+	PreaggPartitionSelection::Builder test_builder;
+  	auto failed_build = test_builder.SetEpsilon(4.5).SetDelta(6.0)
+  									.SetMaxPartitionsContributed(7).Build();
+  	EXPECT_THAT(failed_build.status().code(),
+                Eq(base::StatusCode::kInvalidArgument));
+   	std::string message(std::string(failed_build.status().message()));
+  	EXPECT_THAT(message, MatchesRegex("^Delta has to be in the interval.*"));
+}
+
+TEST(PartitionSelectionStrategiesTest,
+	 PreaggPartitionSelectionUnsetMaxPartitionsContributed) {
+	PreaggPartitionSelection::Builder test_builder;
+  	auto failed_build = test_builder.SetEpsilon(0.8).SetDelta(0.9).Build();
+  	EXPECT_THAT(failed_build.status().code(),
+                Eq(base::StatusCode::kInvalidArgument));
+   	std::string message(std::string(failed_build.status().message()));
+  	EXPECT_THAT(message, MatchesRegex("^Max number of partitions a user can"
+  									  " contribute to has to be set.*"));
+}
+
+TEST(PartitionSelectionStrategiesTest,
+	 PreaggPartitionSelectionNegativeMaxPartitionsContributed) {
+	PreaggPartitionSelection::Builder test_builder;
+  	auto failed_build = test_builder.SetEpsilon(0.1).SetDelta(0.2)
+  									.SetMaxPartitionsContributed(-3).Build();
+  	EXPECT_THAT(failed_build.status().code(),
+                Eq(base::StatusCode::kInvalidArgument));
+   	std::string message(std::string(failed_build.status().message()));
+  	EXPECT_THAT(message, MatchesRegex("^Max number of partitions a user can"
+  									  " contribute to has to be positive.*"));
+}
+
+//We expect the probability of keeping a partition with one user
+//will be approximately delta
+TEST(PartitionSelectionStrategiesTest, PreaggPartitionSelectionOneUser) {
+	PreaggPartitionSelection::Builder test_builder;
+	std::unique_ptr<PartitionSelectionStrategy> build =
+		test_builder.SetEpsilon(0.5).SetDelta(0.02)
+					.SetMaxPartitionsContributed(1).Build().ValueOrDie();
+  	double num_kept = 0.0;
+  	for(int i = 0; i < 1000000; i++) {
```

Comment:
If you're using the same number of samples everywhere make it a file-level constant. If you want to use a unique number of samples for a particular test, make it a local variable in that test (so you don't have to worry about keeping the loop and the probability calculation in sync).

Reply:
There were two values that were used repeatedly - 10^6 and 10^7.  I made two constants...O__o hope that's okay